src/relay/qnn/op/mul.cc

```diff
@@ -71,31 +59,35 @@ Expr QnnMulCanonicalize(const Attrs& attrs, const Array<Expr>& new_args,
   which is essentially a requantization of tensor Q' into tensor Q_c.
   */
 
-  auto lhs_shifted = Cast(lhs, DataType::Int(32));
-  auto rhs_shifted = Cast(rhs, DataType::Int(32));
+  auto lhs_shifted = Cast(args.lhs, fullPrecisionInt32);
```

Comment:
I think DataType::Int(32) is more or as informative as fullPrecisionInt32.

Reply:
do you think `int32Dtype` would be better? using DataType::Int(32) could be an error waiting in happening, if we forget to change it someplace.

Reply:
I understand the concern. At the same time, one might want to upcast only to int16 and not int32 while subtracting zero points for one operator, while keep int32 for other (for example, in pool, even int16 upcasting is ok). Similarly, one might choose int64 for mul in future. So, I think it is safe to keep DataType::Int(32).

Reply:
One compromise could be just to name them `Int32` or `Float32` which carries the necessary information and also has the safety associated. For now I have made the change have added the names Int32 and Float32 so it both intuitive and easy to change.